### PR TITLE
ref(ts) Convert TimeSelector to typescript

### PR DESCRIPTION
**Pull request by markstory**
A few types needed to be massaged to align with types elsewhere. Notably the parameters of getPeriodAgo() needed to be inverted to align with the definition of those functions.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/index.tsx [hunk 1]
```diff
@@ -23,68 +23,70 @@ import {
...
+  /**
+   * Should we have a time selector?
+   */
+  showTimePicker?: boolean;
+
+  /**
+   * Use UTC
+   */
+  utc?: boolean | null;
+} & typeof defaultProps;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/index.tsx:75]: Do you know if the the docblocks are carried over from `defaultProps` when doing this?

It always bothers me a bit that we don't declare all of the types in `Props` when they get sucked in from defaultProps.

Comment (markstory) [reply src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/index.tsx:75]: I don't think they do :cry:

**Review by matejminar — APPROVED** [review]

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/index.tsx (hunk 1 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/index.tsx:75]: We could not use typeof and go with this first approach here:
https://develop.sentry.dev/frontend/defaultprops/#class-components

That should carry over dockblocks. Still does not work well with HoCs I'm afraid.

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/index.tsx [hunk 2]
```diff
@@ -58,84 +58,94 @@ const SelectorItemsHook = HookOrDefault({
...
+  /**
+   * Start date value for absolute date selector
+   */
+  start: DateString;
+
+  /**
+   * End date value for absolute date selector
+   */
+  end: DateString;
+
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/organizations/timeRangeSelector/index.tsx:89]: should we remove these empty lines to be consistent?

**Review by priscilawebdev — APPROVED** [review]
Looks good to me

**Pull request merged**